src/actions/errors/show.cr

```diff
@@ -1,52 +1,60 @@
 class Errors::Show < Lucky::ErrorAction
-  def handle_error(error : JSON::ParseException)
-    message = "There was a problem parsing the JSON." +
-              " Please check that it is formed correctly"
+  DEFAULT_MESSAGE = "Something went wrong."
+  default_format :html
+  dont_report [Lucky::RouteNotFoundError]
 
-    if json?
-      json Errors::ShowSerializer.new(message), status: 400
+  def render(error : Lucky::RouteNotFoundError)
+    if html?
+      error_html "Sorry, we couldn't find that page.", status: 404
     else
-      render_error_page status: 500
+      error_json "Not found", status: 404
     end
   end
 
-  def handle_error(error : Lucky::RouteNotFoundError)
-    if json?
-      json Errors::ShowSerializer.new("Not found"), status: 404
+  # When the request is JSON and an InvalidOperationError is raised, show a
+  # helpful error with the param that is invalid, and what was wrong with it.
+  def render(error : Avram::InvalidOperationError)
+    if html?
+      error_html DEFAULT_MESSAGE, status: 500
     else
-      render_error_page title: "Sorry, we couldn't find that page.", status: 404
+      error_json \
+        message: error.renderable_message,
+        details: error.renderable_details,
+        param: error.invalid_attribute_name,
+        status: 400
     end
   end
 
-  # This is the catch all method that renders unhandled exceptions
-  def handle_error(error : Exception)
-    Lucky.logger.error(unhandled_error: error.inspect_with_backtrace)
-
-    if Lucky::ErrorHandler.settings.show_debug_output
-      # In development and test, render a debug page
-      render_detailed_exception_page(error)
+  # Always keep this below other 'render' methods or it may override your
```

Comment:
Interesting I would have thought it would override if it were below other render methods, i.e. the last defined version of a method overrides all previous definitions.

Reply:
Actually you might be right. Going to open an issue to investigate this. https://github.com/luckyframework/lucky_cli/issues/422